.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,58 @@
+name: Build and Deploy Docs
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build_docs:
+    name: Build and Deploy Documentation
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # to ensure other branches are available for build
+    - name: Install system dependencies
+      run: sudo apt-get install -y libhdf5-dev
+    - name: Set r compilation options
+      run: bash -c 'echo -e "MAKEFLAGS=--jobs=3\nCXXFLAGS += -O1 -UNDEBUG" > "$GITHUB_WORKSPACE/Makevars.user" && echo "R_MAKEVARS_USER=$GITHUB_WORKSPACE/Makevars.user" >> "$GITHUB_ENV"'
+    - name: Setup R
+      uses: r-lib/actions/setup-r@v2
+    - name: Install R dependencies
+      uses: r-lib/actions/setup-r-dependencies@v2
+      with: 
+        cache-version: 1
+        working-directory: 'r'
+        extra-packages: |
+            any::pkgdown
+            any::devtools
+    - name: Get last commit message
+      id: last_commit
+      run: |
+        {
+          echo 'LAST_COMMIT_MESSAGE<<EOF'
+          git log -n 1
+          echo 'EOF'
+        } >> "$GITHUB_ENV"
+    - name: Create git worktree with docs branch
+      run: git worktree add r/docs docs-html
+    - name: Build documentation
+      run: |
+        Rscript .github/workflows/deploy-reference.R
+    - name: Configure Git
+      run: |
+        git config --global user.name "${{ github.actor }}"
```

Comment:
Choice to create the documentation commit using the person who created the PR.  We can change this behaviour to the `gh actions` worker instead though